internal/pkg/manifest/validate.go

```diff
@@ -745,36 +765,54 @@ func (CommandOverride) validate() error {
 	return nil
 }
 
+// validate returns nil if RoutingRuleConfiguration is configured correctly.
+func (r RoutingRuleConfiguration) validate() error {
+	if r.IsEmpty() {
+		return nil
+	}
+	// we consider the fact that primary routing rule is mandatory before you write any additional routing rules.
+	if err := r.MainRoutingRule.validate(); err != nil {
+		return err
+	}
+	// checking this condition here as validate method on ALBRoutingRule only validates common parameters between
+	// MainRoutingRule and AdditionalRoutingRules
+	if r.MainRoutingRule.TargetContainer != nil && r.TargetContainerCamelCase != nil {
+		return &errFieldMutualExclusive{
+			firstField:  "target_container",
+			secondField: "targetContainer",
+		}
+	}
+
```

Comment:
Do you think we can do something like this to motivate users to move away from the deprecated field: if targetContainer is specified with additional_rules, we error out and ask users to move to  target_container

Reply:
As per discussion with Wanxian, I will address this comment separately in another PR. Feel free to review and leave your comments. 

cc: @Lou1415926 @efekarakus